### ISSUE TITLE
Add a new parse error when encountering lifetimes in inappropriate places

### ISSUE THREAD

**Issue by estebank** [rust-lang/rust#48899]
[The following code](https://play.rust-lang.org/?gist=252a4b0cfca62aa5aa272afd013b11dc&version=nightly) should complain about the lifetime passed to `foo`:
```
fn foo<'a>(bar: &'a usize) {}

fn main() {
    let bar = 3;
    foo(&'a bar);
}
```

instead, it complains about `bar`:

```
error: expected `:`, found `bar`
 --> src/main.rs:5:13
  |
5 |     foo(&'a bar);
  |             ^^^ expected `:`
```

**Comment by Centril** [comment]
Now we have:
```rust
error: borrow expressions cannot be annotated with lifetimes
 --> src/main.rs:5:9
  |
5 |     foo(&'a bar);
  |         ^--^^^^
  |          |
  |          annotated with lifetime here
  |          help: remove the lifetime annotation
```
